pkg/ride/environment.go

```diff
@@ -751,6 +753,28 @@ func (ws *WrappedState) incrementInvCount() {
 	ws.invokeCount++
 }
 
+func (ws *WrappedState) validateBalances() error {
+	for key, balanceDiff := range ws.diff.balances {
+		address := proto.NewRecipientFromAddress(key.address)
+		var (
+			balance uint64
+			err     error
+		)
+		if key.asset.Present {
+			balance, err = ws.diff.state.NewestAssetBalance(address, key.asset.ID)
+		} else {
+			balance, err = ws.diff.state.NewestWavesBalance(address)
+		}
+		if err != nil {
+			return err
+		}
+		if int64(balance)+balanceDiff.regular < 0 {
```

Comment:
Here better to check for int64 overflow, as it made in state.

Reply:
check for an overflow and negative balance, right?